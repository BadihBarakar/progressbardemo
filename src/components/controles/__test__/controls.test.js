import React from "react";
import ReactDOM from "react-dom";
import Controls from "../controls";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

// mock the handling funtions for test purposes...
const handleButton = jest.fn(() => {
  return true;
});

const handleSelectBar = jest.fn(() => {
  return true;
});

it("render controls successfully", () => {
  // test that the component will render successfully in another container...
  const endPointParams = {
    buttons: [14, 20, -35, -11],
    bars: [57, 66],
    limit: 190,
  };

  const div = document.createElement("div");
  ReactDOM.render(
    <Controls
      buttons={endPointParams.buttons ? endPointParams.buttons : null}
      bars={endPointParams.bars ? endPointParams.bars.length : null}
      handleButtonClick={handleButton}
      onSelectChange={handleSelectBar}
    />,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders controls correctly", () => {
  // test that the component contains all correct elements required...
  const endPointParams = {
    buttons: [5, 46, -32, -22],
    bars: [40, 22, 87, 50],
    limit: 160,
  };

  const component = render(
    <Controls
      buttons={endPointParams.buttons ? endPointParams.buttons : null}
      bars={endPointParams.bars ? endPointParams.bars.length : null}
      handleButtonClick={handleButton}
      onSelectChange={handleSelectBar}
    />
  );
  expect(component.getByTestId("mainDiv"));
  expect(component.getByTestId("selectDiv"));
  expect(component.getByTestId("selectProgBar"));
  expect(component.getByTestId("selectProgBar"));
  for (let n = 0; n < endPointParams.buttons.length; n++) {
    expect(component.getByTestId("buttonDiv" + n));
    expect(component.getByTestId("button" + n));
  }
});

it("match snapshot", () => {
  // test that the component matches the snapshot,
  // creating a snpshot if it doesn't exist...
  const endPointParams = {
    buttons: [19, 6, -27, -27],
    bars: [13, 19, 10, 86],
    limit: 200,
  };
  const tree = renderer.create(
    <Controls
      buttons={endPointParams.buttons ? endPointParams.buttons : null}
      bars={endPointParams.bars ? endPointParams.bars.length : null}
      handleButtonClick={handleButton}
      onSelectChange={handleSelectBar}
    />
  );
  expect(tree).toMatchSnapshot();
});
