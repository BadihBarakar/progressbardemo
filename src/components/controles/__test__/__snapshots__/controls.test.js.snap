// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="controls-container"
  data-testid="mainDiv"
>
  <div
    className="select-div"
    data-testid="selectDiv"
  >
    <select
      className="prog-select"
      data-testid="selectProgBar"
      id="selProgBar"
      onChange={[MockFunction]}
    >
      <option
        value={0}
      >
        #progress 
        1
      </option>
      <option
        value={1}
      >
        #progress 
        2
      </option>
      <option
        value={2}
      >
        #progress 
        3
      </option>
      <option
        value={3}
      >
        #progress 
        4
      </option>
    </select>
  </div>
  <div
    className="button-div"
    data-testid="buttonDiv0"
  >
    <button
      className="control-button"
      data-testid="button0"
      onClick={[Function]}
    >
      19
    </button>
  </div>
  <div
    className="button-div"
    data-testid="buttonDiv1"
  >
    <button
      className="control-button"
      data-testid="button1"
      onClick={[Function]}
    >
      6
    </button>
  </div>
  <div
    className="button-div"
    data-testid="buttonDiv2"
  >
    <button
      className="control-button"
      data-testid="button2"
      onClick={[Function]}
    >
      -27
    </button>
  </div>
  <div
    className="button-div"
    data-testid="buttonDiv3"
  >
    <button
      className="control-button"
      data-testid="button3"
      onClick={[Function]}
    >
      -27
    </button>
  </div>
</div>
`;
